### ISSUE TITLE
unit tests failed after install -- problem with integrator

### ISSUE THREAD

**Issue by pnave95** [brian-team/brian2#1005]
I am running Ubuntu 16.04 LTS.

I just completed fresh installs of Anaconda (python 3.6.6) and brian2 (v2.7) into a brand new anaconda environment.  I ran brian2.test() after install and was confronted 3/3 failed tests.  I then ran brian2.test('numpy') with the result of 2/2 failed tests.  

After running "brian2.test('numpy')", the the first failed test seemed to be coming from a problem with some linear integrator (error posted below).  

```
==============================================================
FAIL: brian2.tests.test_stateupdaters.test_priority
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/patrick/anaconda3/envs/brian2env/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/patrick/anaconda3/envs/brian2env/lib/python3.6/site-packages/brian2/tests/test_stateupdaters.py", line 364, in test_priority
    raise AssertionError('Should not be able to integrate these '
AssertionError: Should not be able to integrate these equations
-----------------------------------------------------------------------
Ran 347 tests in 34.588s

FAILED (SKIP=4, failures=1)
```

Before this, I was warned "WARNING:  Integrating equations with GSL is still considered experimental [brian2.stateupdaters.GSL]", but this did not cause an explicit error.  I am not sure if this is related, but I suspect it may well be.

I am hoping the later errors are related effects of this first error.  I would appreciate any help in resolving this issue.  Thanks!

**Comment by mstimberg** [comment]
Hi. This is an issue with the sympy version that has been released after our last Brian 2 release. We already fixed this issue in the master branch and the fix will be included in the next release which will happen in the coming week. 

It is mostly an issue with our tests, though, in the vast majority of cases you should be able to use Brian just fine despite this test failure.

**Comment by pnave95** [comment]
@mstimberg I have not tried the update yet, but there seems to be more than just a test problem with the version I downloaded.  Running through some simple Hodgkin-Huxley models, there are times when exponential_euler method seems to work fine, but the 4th order runge-kutta (rk4) method generates warnings and produces spurious results.  This is highly suspicious since runge-kutta methods should be substantially more accurate than euler based methods.

**Comment by mstimberg** [comment]
RK4 is more accurate but less stable than exponential Euler, you'll have to use a small time step on the order of 0.01ms. See e.g. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3779145/

**Comment by mstimberg** [comment]
The incompatibility with recent sympy version is fixed with the latest Brian release (2.2).
